File: backend/Entities/ClinicSettings.cs
namespace ClinicFlow.Api.Entities;

/// <summary>
/// Single-row clinic configuration. There is only ever one settings record.
/// </summary>
public class ClinicSettings
{
    public Guid Id { get; set; }
    public required string ClinicName { get; set; }
    public string? PhoneNumber { get; set; }
    public string? Email { get; set; }
    public string? Address { get; set; }
    public TimeOnly? OpeningTime { get; set; }
    public TimeOnly? ClosingTime { get; set; }
    public required string DefaultCurrency { get; set; }
    public DateTime CreatedAtUtc { get; set; }
    public DateTime? UpdatedAtUtc { get; set; }
}

File: frontend/src/types/clinicSettings.ts
export interface ClinicSettings {
  id: string;
  clinicName: string;
  phoneNumber: string | null;
  email: string | null;
  address: string | null;
  /** "HH:mm" form, matching an HTML time input. */
  openingTime: string | null;
  closingTime: string | null;
  defaultCurrency: string;
  createdAtUtc: string;
  updatedAtUtc: string | null;
}

export interface UpdateClinicSettingsRequest {
  clinicName: string;
  phoneNumber?: string;
  email?: string;
  address?: string;
  openingTime?: string;
  closingTime?: string;
  defaultCurrency: string;
}
